core/vm/logger.go

```diff
@@ -98,6 +99,34 @@ func (s *StructLog) ErrorString() string {
 	return ""
 }
 
+// StructFrame is emitted to the EVM upon stepping into a new call frame.
+type StructFrame struct {
```

Comment:
Maybe just `Frame` or `Scope`? I never liked `StructLog` anyway. Does it need to be public? 
Why does it need special marshalling rules, like the `GasUsed` being json-special-cased?

Reply:
Happy with either. Because I used one `Frame` object to capture a complete frame (enter and exit info) but didn't want to log empty `gasUsed` and `output` fields in the json logger.

Do you think the `StructLogger` needs to capture `enter` and `exit` or should I remove it here too?

Reply:
I had a cursory look and couldn't find where scope info would be useful for `StructLogger` so removed `StructFrame` and co.